tests/test_private_attributes.py

```diff
@@ -289,7 +289,7 @@ class Model(ParentAModel, ParentBModel):
 def test_private_attributes_not_dunder() -> None:
     with pytest.raises(
         NameError,
-        match="Private attributes must not use dunder names;" " use a single underscore prefix instead of '__foo__'.",
+        match='Private attributes must not use dunder names;' " use a single underscore prefix instead of '__foo__'.",
```

Comment:
Clearly it's having a hard time with this as well 😆

Reply:
Yea that is weird. I wonder what it's tripping on...

Reply:
So, what's happening here is that it's trying to convert each segment to single quotes (as per `quote-style = 'single'`). It's succeeding for the first segment in this implicit concatenation, but it avoids changing the second segment, because that segment itself _contains_ a single quote (and so that single quote would need to be escaped).

Arguably we should be making that decision across the _entire_ implicit concatenation (i.e., looking across all segments) to avoid these mixed quotes. I'll file an issue and we'll discuss.

Reply:
Ruff changes the quotes of the first string but not of the second because it would require escaping the `'`.  Is it unexpected that Ruff uses different quotes for different parts of the implicit concatenated string? Would you prefer if it would change the quotes consistently for the entire implicit concatenated string?

Reply:
Ah, thanks for the explanation. Clearly I missed the pattern of escaping `'` that was at the root of many of my questions.

I do feel like changing quotes consistently for an entire implicit concatenated string might look a bit cleaner, but I'm not adamant about the change. Excited to hear what others think. 

@charliermarsh, thanks for filing an issue 👍

Reply:
@charliermarsh could you link the issue here? I may have just missed it, but didn't see it at a first glance

Reply:
Apologies -- just filed it here: https://github.com/astral-sh/ruff/issues/8265